go/libkb/merkle_client.go

```diff
@@ -1159,7 +1241,35 @@ func (vp *VerificationPath) verifyUser(ctx context.Context) (user *MerkleUserLea
 
 	user, err = parseMerkleUserLeaf(ctx, leaf, vp.G())
 	if user != nil {
-		user.uid = vp.uid
+		user.uid = vp.userInfo.uid
+	}
+	return
+}
+
+func (vp *VerificationPath) verifyTeam(ctx context.Context, teamID keybase1.TeamID) (teamLeaf *MerkleTeamLeaf, err error) {
+	curr := vp.root.RootHash()
+
+	var leaf *jsonw.Wrapper
+	leaf, err = vp.path.VerifyPath(curr, teamID.String())
```

Comment:
Yeah if `err` is a `MerkleNotFoundError` then `verifyTeam` ends up returning `&MerkleTeamLeaf{}, nil`. Maybe the should be more explicit, or different. For the user side of things somehow the whole system depends on that, I tried changing it. It has to do with loading yourself when signing up, and nuked users.